### ISSUE TITLE
podman cp support

### ISSUE THREAD

**Issue by mairin** [containers/podman-desktop#2801]
### Is your enhancement related to a problem? Please describe

I was working on containerizing an app and realized the default config file shipped with the base image (was for php-fpm) had a config i needed to modify... I need a copy of the config file out of the container in order to make a custom version of it and pull it into the Containerfile via wget when I build the image. 

Anyhow, I needed a copy of the config out of the container in order to be able to do this, and I discovered podman cp which made this very easy. 

### Describe the solution you'd like

Not sure how possible this might be, but it might be neat to have a list of files / most recently-visited files / etc. from the container terminal, and you could download them locally from the Podman Desktop client (via podman cp)

### Describe alternatives you've considered

I tried copy/pasting screen-by-screen and apparently made a copy paste error, so that sucked and didnt work :)

podman cp on the cmdline is fine but slicker to do via the UI... i didn't even know podman cp existed.

**Comment by afbjorklund** [comment]
The canonical way to do this was `podman mount` (it took a _long_ time to get `podman cp` in),
but I guess that would only get you "half the way" - it would mount the container to the VM...

So you would need to make sure to also have some path that is being shared with the host.
But when all of that is out of the way, you should be able to copy or edit files in the normal way?

----

https://docs.podman.io/en/latest/markdown/podman-cp.1.html#alternatives

https://docs.podman.io/en/latest/markdown/podman-machine-init.1.html#volume-v-source-target-options

EDIT: I guess that old method (with mount) isn't very convenient to use anymore, anyway.

```console
[anders@lima-podman anders]$ podman mount nice_wescoff
Error: cannot run command "podman mount" in rootless mode, must execute `podman unshare` first
[anders@lima-podman anders]$ podman unshare
[root@lima-podman anders]# podman mount nice_wescoff
/home/anders.linux/.local/share/containers/storage/overlay/f279bb0b3c8be92a4c6d01b7e4a88994cd9d7bdeac5d87c57bcfddbb3b5fc888/merged
```

**Comment by afbjorklund** [comment]
While it is nice that one can use `cp` and `ssh`, it is more convenient to set up https://mutagen.io

Something like https://mutagen.io/documentation/orchestration/projects should set it up for you

**Comment by feloy** [comment]
Reopening, for another usage:

An extension may want to start a container based on a generic image (httpd, postgres, ...), with some specific configuration file.

An alternative is to create a new image including this configuration file, but as the config can contain sensitive information (password, etc), there is a risk that the user pushes the image to a registry and exposes the sensitive information.

A convenient possibility would be to start the container with the generic image and to mount a volume containing this file. But we need to copy the config file to the volume first, and an equivalent of `podman cp` would be useful for that.

**Comment by benoitf** [comment]
you don't want to write the file to a folder on the host and mount that file when starting the container ? so it's persisted

This is how postgres is teaching people:
https://github.com/docker-library/docs/blob/master/postgres/README.md#database-configuration

Sensitive information is also often provided with environment variables
`-e POSTGRES_PASSWORD=mysecretpassword`

**Comment by feloy** [comment]
Yes, that would be a possibility, and is probably the best one when the user manages himself the container and the user wants to persist the files between several container creation, but in the case of an extension which would create these configuration files on the fly, the extension would have to manage these files in the host, to clean them up when the container has been deleted.

As a concrete example, let's suppose an extension, having information to connect to a Postgres database, and which would like to start a pgadmin4 container connected to this database. The configuration files are described in this thread (https://stackoverflow.com/questions/66578506/where-is-the-pgpass-file-in-pgadmin4-docker-container-when-this-file-is-mounted/70040661#70040661). Two files (pgpass and servers.json) need to be created on the fly, and mounted in the container in some way (ignore the compose.yaml file, which would be handled by the extension itself).

**Comment by afbjorklund** [comment]
The config volume should be cleaned up with the container, unless the user has specifically named it (and thus want to keep it)

**Comment by feloy** [comment]
> The config volume should be cleaned up with the container, unless the user has specifically named it (and thus want to keep it)

The volume would be created by the extension, not by the user. But right, the volume would need to be cleaned up too, but that seems more reasonable to me, as it is easier to detect that a volume is in use or not, compared to detecting that some files in the FS of the user are mounted or not

**Comment by benoitf** [comment]
The thing that I'm not sure to fully follow is:

> An extension may want to start a container based on a generic image (httpd, postgres, ...), with some specific configuration file.
> An alternative is to create a new image including this configuration file, but as the config can contain sensitive information (password, etc), there is a risk that the user pushes the image to a registry and exposes the sensitive information.

I don't understand the "generic image" matching with the "specific configuration file"

What is a real use-case of an extension wanted to start an httpd server and then a postgres server ? that seems very different ?
so how can it provide the good configuration file if it's too generic ?

if it's managed by an extension, I would assume that the extension would just provide sensitive data with environment variables.

>  Two files (pgpass and servers.json) need to be created on the fly, and mounted in the container in some way (ignore the compose.yaml file, which would be handled by the extension itself).

but why the extension would not push a custom image without the sensitive information ?

**Comment by feloy** [comment]
> The thing that I'm not sure to fully follow is:
> 
> > An extension may want to start a container based on a generic image (httpd, postgres, ...), with some specific configuration file.
> > An alternative is to create a new image including this configuration file, but as the config can contain sensitive information (password, etc), there is a risk that the user pushes the image to a registry and exposes the sensitive information.
> 
> I don't understand the "generic image" matching with the "specific configuration file"
> 
> What is a real use-case of an extension wanted to start an httpd server and then a postgres server ? that seems very different ? so how can it provide the good configuration file if it's too generic ?
> 

That was two different examples. An 'nginx' extension may want to start the `docker.io/nginx` image with some specific configuration, or a postgres extension may want to start the `docker.io/postgres` image with some specific configuration.

> if it's managed by an extension, I would assume that the extension would just provide sensitive data with environment variables.
> 

It's not always possible. For example for the postgres image, the extension may want to add some scripts (https://github.com/docker-library/docs/blob/master/postgres/README.md#initialization-scripts), and these scripts may  contain sensitive information.

> > Two files (pgpass and servers.json) need to be created on the fly, and mounted in the container in some way (ignore the compose.yaml file, which would be handled by the extension itself).
> 
> but why the extension would not push a custom image without the sensitive information ?
